app/controllers/posts_controller.rb

```diff
@@ -299,7 +299,7 @@ def all_reply_ids
 
   def destroy
     post = find_post_from_params
-    unless current_user.staff?
+    unless current_user.staff? || guardian.is_category_group_moderator?(post.topic.category)
```

Comment:
maybe we use a guardian here? Then we get to reuse in recover.  

`if !guardian.can_moderate?(post)`